### PR TITLE
test: eigenpod unit tests with checkpointing

### PR DESCRIPTION
**Pull request by 8sunyuan**
Fixed the EigenPod unit tests in the `src/test/unit/EigenPodUnit.t.sol` file.
Uses the mocked beaconChain helper from integration tests as well as a `EigenPodUser` to mock beaconChain state.

Additional tests using proofs from our backend proofgen would be good to add when implemented.

### CODE REVIEW

File: src/contracts/libraries/BeaconChainProofs.sol [hunk 1]
```diff
@@ -289,7 +289,7 @@ library BeaconChainProofs {
     }
 
     /// @dev Retrieves a validator's effective balance (in gwei)
-    function getEffectiveBalanceGwei(bytes32[] memory validatorFields) internal pure returns (uint64) {
+    function getEffectiveBalanceGwei(bytes32[] memory validatorFields) public pure returns (uint64) {
```
Comment (wadealexc) [inline src/contracts/libraries/BeaconChainProofs.sol:292]: Why was this changed to public? We probably don't want that -- this will resolve to a `delegatecall`, pretty sure.

File: src/contracts/pods/EigenPod.sol [hunk 2]
```diff
@@ -32,7 +33,8 @@ contract EigenPod is
     Initializable, 
     ReentrancyGuardUpgradeable, 
     EigenPodPausingConstants, 
-    EigenPodStorage 
+    EigenPodStorage,
+    Test
```
Comment (wadealexc) [inline src/contracts/pods/EigenPod.sol:37]: Oh, we should definitely remove Test artifacts!

File: src/contracts/pods/EigenPod.sol [hunk 3]
```diff
@@ -409,13 +410,11 @@ contract EigenPod is
 
     /**
      * @notice internal function that proves an individual validator's withdrawal credentials
-     * @param beaconTimestamp is the timestamp whose state root the `proof` will be proven against.
      * @param validatorIndex is the index of the validator being proven
      * @param validatorFieldsProof is the bytes that prove the ETH validator's  withdrawal credentials against a beacon chain state root
      * @param validatorFields are the fields of the "Validator Container", refer to consensus specs
      */
     function _verifyWithdrawalCredentials(
-        uint64 beaconTimestamp,
```
Comment (wadealexc) [inline src/contracts/pods/EigenPod.sol:418]: Oh, nice catch! Did not realize this was unused.

File: src/contracts/libraries/BeaconChainProofs.sol (hunk 1 above)
Comment (8sunyuan) [reply src/contracts/libraries/BeaconChainProofs.sol:292]: Ahh this was needed for a test but I must have mistaken it for being in the BeaconChainMock file

File: src/contracts/pods/EigenPod.sol (hunk 2 above)
Comment (8sunyuan) [reply src/contracts/pods/EigenPod.sol]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/591/commits/20bf8100b15a60ae01f3319eac1a97ebea1a13b7

File: src/contracts/libraries/BeaconChainProofs.sol (hunk 1 above)
Comment (8sunyuan) [reply src/contracts/libraries/BeaconChainProofs.sol]: https://github.com/Layr-Labs/eigenlayer-contracts/pull/591/commits/20bf8100b15a60ae01f3319eac1a97ebea1a13b7

File: src/test/unit/EigenPodUnit.t.sol [hunk 4]
```diff
@@ -88,23 +101,231 @@ contract EigenPodUnitTests is EigenLayerUnitTestSetup {
...
+
+        stakerName = string.concat("Staker", numStakers.toString());
+        staker = new EigenPodUser(stakerName);
+
+        uint256 amount = bound(rand, 1 ether, 640 ether);
+        cheats.deal(address(staker), amount);
+
+        numStakers++;
+        return (staker, amount);
+    }
```
Comment (wadealexc) [inline src/test/unit/EigenPodUnit.t.sol:138]: Probably a decent simplification here -- have this method return `(EigenPodUser, EigenPod)`. I noticed that you aren't using the `uint` return value anywhere, and in most places you call this, you then spend an extra line after getting the pod.

**Review by wadealexc — APPROVED** [review]
Looks great :)

Some additional work to be done:
- cleaning up verifyCheckpointProofs tests
- adding additional checks when completing checkpoints (specifically checking more state changes)
